### PR TITLE
Support Rails 8.0

### PR DESCRIPTION
**Pull request by olivier-thatch**
* Support Rails 8.0: https://guides.rubyonrails.org/8_0_release_notes.html

### CODE REVIEW

**Comment by olivier-thatch** [comment]
https://github.com/rails/rails/pull/52428 changed the API for DB adapters, so the patches will have to be updated accordingly.

The good news is that the unified interface should make patching easier. The bad news is that SaferRailsConsole will have to use different patches for Rails < 8.0 and >= 8.0.

I'll take another stab at this later.

**Comment by olivier-thatch** [comment]
@erikkessler1 @atsheehan Rails 8.0 was released a few days ago. This PR is ready for review.

File: lib/safer_rails_console/patches/sandbox/auto_rollback.rb [hunk 1]
```diff
@@ -35,10 +47,24 @@ def execute_and_clear(...)
         end
 
         if defined?(::ActiveRecord::ConnectionAdapters::PostgreSQLAdapter)
-          ::ActiveRecord::ConnectionAdapters::PostgreSQLAdapter.prepend(PostgreSQLAdapterPatch)
+          if SaferRailsConsole::RailsVersion.eight_or_above?
+            ::ActiveRecord::ConnectionAdapters::PostgreSQLAdapter.prepend(PostgreSQLAdapteRails8Patch)
+          elsif SaferRailsConsole::RailsVersion.six_or_above?
+            ::ActiveRecord::ConnectionAdapters::PostgreSQLAdapter.prepend(PostgreSQLAdapteRails67Patch)
+          end
```
Comment (erikkessler1) [inline lib/safer_rails_console/patches/sandbox/auto_rollback.rb:54]: I'm worried that `8Patch` won't make sense when we have Rails 9+, so what do you think about calling the 8+ patch, `PostgreSQLAdapterPatch` and the old one, `LegacyPostgreSQLAdapterPatch`?

**Review by erikkessler1 — CHANGES_REQUESTED** [review]
Thank you @olivier-thatch! Looks good overall, just a couple of things to address in-line.

File: .circleci/config.yml [hunk 2]
```diff
@@ -82,11 +82,11 @@ workflows:
           matrix:
             parameters:
               ruby_version:
-                - 3.1.4
```
[the remark this answers is not in the document]
Comment (olivier-thatch) [reply .circleci/config.yml:85]: 👍  For some reason I thought Rails 8 was dropping support for 3.1, but that doesn't seem to be the case. Fixed.

File: lib/safer_rails_console/patches/railtie/sandbox.rb [hunk 3]
```diff
@@ -30,6 +30,7 @@ def start(*args)
 end
 
 if SaferRailsConsole::RailsVersion.supported?
+  require 'rails/command'
```
[the remark this answers is not in the document]
Comment (olivier-thatch) [reply lib/safer_rails_console/patches/railtie/sandbox.rb]: I added it to fix an error during development, but it doesn't seem to be happening anymore. Either it was on my end or it was fixed between the RC and the final release. In any case, I've removed it.

File: lib/safer_rails_console/patches/sandbox/auto_rollback.rb (hunk 1 above)
Comment (olivier-thatch) [reply lib/safer_rails_console/patches/sandbox/auto_rollback.rb]: 👍  Good idea, done!

File: .circleci/config.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (olivier-thatch) [reply .circleci/config.yml:85]: Oh actually it is the case. From https://edgeguides.rubyonrails.org/upgrading_ruby_on_rails.html#upgrading-from-rails-7-2-to-rails-8-0:

> Rails 8.0 requires Ruby 3.2.0 or newer.

@erikkessler1 What's your preference here? We can drop support for 3.1 entirely as I did initially, or just tweak CI to omit the (Ruby 3.1, Rails 8.0) combination.

[the remark this answers is not in the document]
Comment (erikkessler1) [reply .circleci/config.yml:85]: @olivier-thatch you can use use the `exclude` option in the test matrix to exclude the incompatible versions. For example: https://github.com/salsify/goldiloader/blob/8cdf77c6006e6e045a6fb684376757f2058e692f/.circleci/config.yml#L92-L94

**Review by erikkessler1 — APPROVED** [review]
Thanks @olivier-thatch! I'll get this released as v0.11.0

**Comment by erikkessler1** [comment]
v0.11.0 should now be available

**Pull request merged**